lib/Onyx.ts

```diff
@@ -594,6 +595,47 @@ function clear(keysToPreserve: OnyxKey[] = []): Promise<void> {
         .then(() => undefined);
 }
 
+function updateSnapshots(data: OnyxUpdate[]) {
+    const snapshotCollectionKey = OnyxUtils.getSnapshotKey();
+    if (!snapshotCollectionKey) return;
+
+    const promises: Array<() => Promise<void>> = [];
+    const finalPromise = Promise.resolve();
+
+    const snapshotCollection = OnyxUtils.getCachedCollection(snapshotCollectionKey);
+
+    Object.entries(snapshotCollection).forEach(([snapshotKey, snapshotValue]) => {
+        // Snapshots may not be present in cache. We don't know how to update them so we skip.
+        if (!snapshotValue) {
+            return;
+        }
+
+        let updatedData = {};
+
+        data.forEach(({key, value}) => {
+            // snapshots are normal keys so we want to skip update if they are written to Onyx
+            if (OnyxUtils.isCollectionMemberKey(snapshotCollectionKey, key)) {
+                return;
+            }
+
+            if (typeof snapshotValue !== 'object' || !('data' in snapshotValue)) {
+                return;
+            }
+
+            const snapshotData = snapshotValue.data;
+            if (!snapshotData || !snapshotData[key]) {
+                return;
+            }
+
+            updatedData = {...updatedData, [key]: lodashPick(value, Object.keys(snapshotData[key]))};
+        });
+
+        promises.push(() => merge(snapshotKey, {data: updatedData}));
+    });
```

Comment:
@luacmartins it was valid point from @blazejkustra. We can iterate over collection of snapshots first and build final object and thus reduce number of `merge` calls

Reply:
Makes sense.